src/renderers/dom/client/ReactMount.js

```diff
@@ -724,6 +729,15 @@ var ReactMount = {
       }
     }
   },
+  /**
+   * Rather than count mounts and unmounts, we query the dom for the DOMProperty.ROOT_ATTRIBUTE_NAME. This is used to
+   * dispose of global event listeners, when the last component is removed/unmounted.
+   * @returns Returns the number of react tagged root elements.
+   * @private
+   */
+  _countReactRootElements: function() {
+    return document.querySelectorAll('[ROOT_ATTR_NAME]').length;
+  },
```

Comment:
> The problem i was trying to avoid by counting is that only works if code unmounts each and every component helping the counter stay accurate.

You must explicitly unmount any nodes via React, simply removing them from DOM is a bug and can have some very nasty side-effects. So if someone has forgotten to unmount via React then they have far bigger issues than listeners not being cleaned up.